### DISCUSSION TITLE
CUETools 检测指南

### DISCUSSION THREAD

**Discussion by giscus** [ZexWoo/Giscus#27] [Announcements]
# CUETools 检测指南

如何检查 CD 抓轨的可信程度？

https://zexwoo.blog/posts/tutorials/cuetools/

**Comment by LinxHex** [comment]
你好，请问有两个Offsetted数值该如何修复呢？

```
[CUETools log; Date: 2023/11/8 19:13:51; Version: 2.2.4]
[CTDB TOCID: bvkon.6T.nCEKoYHUNsIYLoxGWU-] found.
Track | CTDB Status
  1   | (243/243) Accurately ripped
  2   | (242/243) Accurately ripped
  3   | (241/243) Accurately ripped
  4   | (241/243) Accurately ripped
  5   | (242/243) Accurately ripped
  6   | (242/243) Accurately ripped
  7   | (241/243) Accurately ripped
  8   | (240/243) Accurately ripped
  9   | (241/243) Accurately ripped
[AccurateRip ID: 000bffe5-005b28a7-78096f09] found.
Track   [  CRC   |   V2   ] Status
 01     [8e499ac0|8bda6983] (00+00/52) No match
 02     [bb9888f5|0bef9096] (00+00/52) No match
 03     [a6e4b620|ce8fe642] (00+00/52) No match
 04     [c7648800|c23f2d57] (00+00/52) No match
 05     [3cd341c0|d0f8fae3] (00+00/52) No match
 06     [058198f9|11cca20c] (00+00/52) No match
 07     [74c63614|3e1600e7] (00+00/52) No match
 08     [c3b5356e|e01a86cb] (00+00/51) No match
 09     [6aaedce3|4a793740] (00+00/51) No match
Offsetted by 30:
 01     [ccac84e7] (07/52) Accurately ripped
 02     [5fa8fe36] (07/52) Accurately ripped
 03     [e36a1446] (07/52) Accurately ripped
 04     [6001150c] (07/52) Accurately ripped
 05     [f0744934] (07/52) Accurately ripped
 06     [9c2d6fc5] (07/52) Accurately ripped
 07     [87a8393e] (07/52) Accurately ripped
 08     [cf135ecc] (07/51) Accurately ripped
 09     [42fba051] (07/51) Accurately ripped
Offsetted by 503:
 01     [49ececa7] (03/52) Accurately ripped
 02     [aa4b950a] (03/52) Accurately ripped
 03     [72f62d7b] (03/52) Accurately ripped
 04     [d73ef816] (03/52) Accurately ripped
 05     [092371ea] (03/52) Accurately ripped
 06     [52b1b2ef] (03/52) Accurately ripped
 07     [204f6b21] (03/52) Accurately ripped
 08     [824eeb05] (03/51) Accurately ripped
 09     [e21c5442] (03/51) Accurately ripped

Track Peak [ CRC32  ] [W/O NULL] 
 --   98.6 [9CCF2F05] [5013D37D]           
 01   44.2 [5AA93B34] [BD9F3ED2]           
 02   93.2 [9370C653] [F1AFCA27]           
 03   78.5 [F0F8BC93] [CAF34A5D]           
 04   78.6 [7739C78F] [AF33BDA1]           
 05   98.6 [34E9AA5D] [6C1ABEEE]           
 06   60.0 [9B4F4A08] [F439AEB7]           
 07   97.4 [1CD8C538] [6FF05D04]           
 08   58.3 [7C708F37] [BEC709F5]           
 09   98.4 [ECE8EC52] [5F8F07CD]  
```

**Reply by ZexWoo** [reply]
30 相对好一些。

其实也没有修复的必要，没有可听差异，如果要分享，可以带这个检测报告。

**Comment by Victor-X-Li** [comment]
非常感谢您的科普！

**Comment by dumbbird** [comment]
很好的科普～最近也在研究CueTools，这里再补充一个小建议。如果想要查看CueTools检测结果的详尽版本，可以开启 [Settings] > [Advanced] > [CTDB] > [Detailed log] = true

这样log里面可以更明确的告诉你这张专辑抓轨的具体版本ID，以及其他版本ID的信息。这有助于比对某张专辑不同版本之间的差异。不知道阁下是否在HDCDAPE论坛里，如果有，我最近也在坛里发了一个关于Cuetools检测结果的讨论，抛砖引玉，欢迎加入讨论！～谢谢。

**Reply by Victor-X-Li** [reply]
您可以发一下您的讨论的网页吗，我对您提到的 Detailed log 很感兴趣！

edit: 去看了一下 HDCD 论坛，发现已经禁止注册了，哎

**Reply by dumbbird** [reply]
好的，我一会把相关讨论的内容转载到一个公开的网站里吧～

**Reply by dumbbird** [reply]
https://discogs.fandom.com/zh/wiki/%E7%94%A8%E6%88%B7%E5%8D%9A%E5%AE%A2:Dumbbird/%E9%80%9A%E8%BF%87CueTools_Log%E6%9D%A5%E5%88%A4%E6%96%AD%E5%90%8C%E5%BC%A0%E4%B8%93%E8%BE%91%E4%B8%8D%E5%90%8C%E7%89%88%E6%9C%AC%E4%B9%8B%E9%97%B4%E7%9A%84%E9%9F%B3%E8%B4%A8%E5%B7%AE%E5%BC%82

具体的抛转引玉的讨论在这里可以阅读。如有任何疑问也可以到fandom与我交流～谢谢

**Reply by Victor-X-Li** [reply]
感谢！我会抽时间自习研究一下的，我这一段时间也对 CUETools 查询 CD 数据很感兴趣！

**Reply by ZexWoo** [reply]
《通过 CueTools Log 来判断同张专辑不同版本之间的音质差异》我已经读完，确实是一个重要且有用的发现。

经过实测，我的结论如下：
1. CTDB TOCID（长编号）确实是以 TOC 为唯一标识的，这也就意味着，完全不同的专辑可能享有同一个 CTDB TOCID。
1. CTDBID（短编号）确实能在一定程度上揭示数据层面的实质性差异，即，但凡音轨数据有实质性不同，就会获得一个不同的 CTDBID。
1. 但是，如果两个版本的专辑只有偏移值的差异，仍然会享有同一个 CTDBID。此时单靠 CTDB 是无法区分两个版本的。例子是《你的名字。》原声带。

另：dumbbird 同志，请给我个邮箱。

**Reply by ZexWoo** [reply]
补充：一些数据极少的 CTDBID 是值得警惕的，因为它们甚至有可能来自盗版光碟的抓轨。

**Comment by dumbbird** [comment]
(1/6) Accurately ripped, or (4/6) differs in 3 samples @00:03:06

另外我也想补充一下针对这种log信息的理解。
遇到这种情况，log的本意其实是说，这首曲目在不同专辑版本中存在两种不同的内容版本。注意那个OR 非常关键。

其中第一个版本（有1个人），是与你当前的这个版本保持一致的。
而另一种版本（有4个人），与你当前的版本存在3个采样单位的差异。

至于是否需要修复，这个还是要具体情况具体分析。如果你比较确定自己抓轨没有问题 + 自己的专辑也是官方正版，那就可能意味着，这张专辑的官方不同版本之间本来就存在差异，这样一来，如果再去用CueTools修复的话，反而是不尊重、手动调整专辑官方发行版本。这也是不推荐的。
